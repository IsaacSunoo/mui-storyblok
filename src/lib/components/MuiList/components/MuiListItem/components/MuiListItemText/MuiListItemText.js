import React from 'react';
import PropTypes from 'prop-types';
import ListItemText from '@material-ui/core/ListItemText';
import Storyblok from '../../../../../../utils/Storyblok';

/**
 * MuiListItemText is used in storyblok redirect to react routes
 */

export const MuiListItemText = ({
  secondary,
  primary,
  rootClass,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <ListItemText
      className={styles.root}
      primary={primary}
      secondary={secondary}
    />
  );
};

export default MuiListItemText;

MuiListItemText.propTypes = {
  /** primary text */
  primary: PropTypes.string.isRequired,
  /** secondary text */
  secondary: PropTypes.string,
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),
};

MuiListItemText.defaultProps = {
  rootClass: [],
  secondary: '',
};
