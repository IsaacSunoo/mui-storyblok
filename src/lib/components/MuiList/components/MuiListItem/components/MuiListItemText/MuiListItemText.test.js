import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import MuiListItemText from './MuiListItemText';

function setup() {
  const props = {
    component: 'MuiListItemText',
    primary: 'primary',
    secondary: 'secondary',
  };
  const comp = shallow(<MuiListItemText {...props} />);
  return { comp, props };
}

describe('<MuiListItemText />', () => {
  it('renders MuiListItemText', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const mockstore = configureMockStore();
    const store = mockstore({});
    const tree = renderer.create((
      <Provider store={store}>
        <MemoryRouter>
          <MuiListItemText {...props} />
        </MemoryRouter>
      </Provider>
    ));
    expect(tree).toMatchSnapshot();
  });
});
