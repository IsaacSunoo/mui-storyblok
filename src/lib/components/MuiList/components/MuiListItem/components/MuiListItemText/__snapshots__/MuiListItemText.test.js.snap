// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiListItemText /> snapshot 1`] = `
<div
  className="MuiListItemText-root makeStyles-root-3 MuiListItemText-multiline"
>
  <span
    className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
  >
    primary
  </span>
  <p
    className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
  >
    secondary
  </p>
</div>
`;
